iothub/service/src/MessageSystemPropertyNames.cs

```diff
@@ -8,40 +8,122 @@ namespace Microsoft.Azure.Devices
     /// </summary>
     public static class MessageSystemPropertyNames
     {
+        /// <summary>
+        /// A user-settable identifier for the messages. If this value is not supplied by the user,
+        /// the service client will set this to a new GUID.
```

Comment:
I think we only do this conditionally, right?

Reply:
I did not get your question.

Reply:
Are you saying this is in client options or something? Let me check the code.